### ISSUE TITLE
[stable/datadog] Please update kube-state-metrics reference

### ISSUE THREAD

**Issue by jeffreymlewis** [helm/charts#22937]
**Describe the bug**
When deploying the latest version of this chart, the following error occurs in the running agent.

```
E0624 22:23:36.443644       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191109102209-3c0d1af94be5/tools/cache/reflector.go:108: Failed to list *v1beta2.VerticalPodAutoscaler: the server could not find the requested resource (get verticalpodautoscalers.autoscaling.k8s.io)
```

This is due to bug https://github.com/kubernetes/kube-state-metrics/issues/1140 which has already been resolved.

Please update [requirements.yaml](https://github.com/helm/charts/blob/master/stable/datadog/requirements.yaml#L3) to a newer version of the `kube-state-metrics` helm chart. Optionally, please allow setting the kube-state-metrics version in values.yaml.

**Version of Helm and Kubernetes**:
```
$ helm3 version
version.BuildInfo{Version:"v3.1.2", GitCommit:"d878d4d45863e42fd5cff6743294a11d28a9abce", GitTreeState:"clean", GoVersion:"go1.13.8"}
```

```
$ kubectl-1.16.11 version --short
Client Version: v1.16.11
Server Version: v1.16.8-eks-e16311
```

**Which chart**:
stable/datadog

**What happened**:
Deployment is fine, but the above errors occur every 10 seconds. 

**What you expected to happen**:
I expect the agent and it's required helm charts to run without error.

**How to reproduce it** (as minimally and precisely as possible):
* run `helm3 install datadog stable/datadog --set datadog.apiKey=<DATADOG_API_KEY>`
* observe resulting errors: `kubectl-1.16.11 logs datadog-kube-state-metrics-XXXXXXXXXX-XXXXXX`

**Anything else we need to know**:
I think that's it. Let me know if you have any questions.

**Comment by JVMartin** [comment]
I don't think this should go stale. Still an issue

**Comment by vboulineau** [comment]
Hello,

As you may have noticed, this repository is deprecated and has been moved to:
https://github.com/DataDog/helm-charts

(More information about the move here: https://github.com/helm/charts/tree/master/stable/datadog)

This issue has been solved in the new versions released from the new repo (`2.4.x`) already.
